### PR TITLE
Core: remove statistic files in CatalogUtil:dropTableData

### PR DESCRIPTION
**Pull request by dramaticlly**
This PR attempt to include puffin stats files as part of CatalogUtil table cleanup

CC @RussellSpitzer @findepi

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/CatalogUtil.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ public static void dropTableData(FileIO io, TableMetadata metadata) {
 
     LOG.info("Manifests to delete: {}", Joiner.on(", ").join(manifestsToDelete));
 
-    // run all of the deletes
+    // run all the deletes
```
Comment (RussellSpitzer) [inline core/src/main/java/org/apache/iceberg/CatalogUtil.java:103]: Probably fine as is?

Comment (dramaticlly) [reply core/src/main/java/org/apache/iceberg/CatalogUtil.java:103]: IntelliJ trying to be smart at syntax so I let it fix but I can revert!

**Review by RussellSpitzer — APPROVED** [review]
Looks good to me

File: core/src/main/java/org/apache/iceberg/CatalogUtil.java [hunk 2]
```diff
@@ -117,6 +117,11 @@ public static void dropTableData(FileIO io, TableMetadata metadata) {
         Iterables.transform(metadata.previousFiles(), TableMetadata.MetadataLogEntry::file),
         "previous metadata",
         true);
+    deleteFiles(
+        io,
+        Iterables.transform(metadata.statisticsFiles(), StatisticsFile::path),
+        "statistic",
```
Comment (ajantha-bhat) [inline core/src/main/java/org/apache/iceberg/CatalogUtil.java:123]:
```suggestion
        "statistics",
```

File: core/src/test/java/org/apache/iceberg/hadoop/TestCatalogUtilDropTable.java [hunk 3]
```diff
@@ -38,19 +49,28 @@
...
 
-    TableMetadata tableMetadata = readMetadataVersion(3);
+    TableMetadata tableMetadata = readMetadataVersion(4);
     Set<Snapshot> snapshotSet = Sets.newHashSet(table.snapshots());
 
     Set<String> manifestListLocations = manifestListLocations(snapshotSet);
     Set<String> manifestLocations = manifestLocations(snapshotSet, table.io());
     Set<String> dataLocations = dataLocations(snapshotSet, table.io());
     Set<String> metadataLocations = metadataLocations(tableMetadata);
+    Set<String> statisticLocations = statisticLocations(tableMetadata);
```
Comment (ajantha-bhat) [inline core/src/test/java/org/apache/iceberg/hadoop/TestCatalogUtilDropTable.java:70]: this variable name and method name can be statisticsLocations ('s' is missing) to be inline with naming of other places or just statsLocations is also fine.

**Review by ajantha-bhat — APPROVED** [review]
LGTM. 

with a minor nit on naming.

File: core/src/main/java/org/apache/iceberg/CatalogUtil.java (hunk 2 above)
Comment (dramaticlly) [reply core/src/main/java/org/apache/iceberg/CatalogUtil.java:123]: thank you @ajantha-bhat , I used `statistic` earlier due to singular nouns here given how all existing code refer to file type, like `manifest`, `manifest list` and `previous metadata`. But I now recognize that the class name is already defined as `api/src/main/java/org/apache/iceberg/StatisticsFile.java`. let me update to your suggestion

**Comment by ajantha-bhat** [comment]
There is also a partition stats file added recently https://github.com/apache/iceberg/commit/6e21bbf4c4cd2c8351a64636f91d05d00492dff2

We should handle this for them aswell. 

I can have a follow up PR, if it is too much to handle in this PR.

File: core/src/test/java/org/apache/iceberg/hadoop/TestCatalogUtilDropTable.java [hunk 4]
```diff
@@ -181,4 +205,34 @@ private Set<String> metadataLocations(TableMetadata tableMetadata) {
     metadataLocations.add(tableMetadata.metadataFileLocation());
     return metadataLocations;
   }
+
+  private Set<String> statsLocations(TableMetadata tableMetadata) {
```
Comment (ajantha-bhat) [inline core/src/test/java/org/apache/iceberg/hadoop/TestCatalogUtilDropTable.java:209]: nit: these two new private methods can be static

**Review by amogh-jahagirdar — APPROVED** [review]
This change looks good @dramaticlly , thanks for the fix! I'll go ahead and merge this.

**Comment by ajantha-bhat** [comment]
> There is also a partition stats file added recently https://github.com/apache/iceberg/commit/6e21bbf4c4cd2c8351a64636f91d05d00492dff2
We should handle this for them aswell.

I will open the follow up PR for partition stats files today.

**Pull request merged**